src/index.ts

```diff
@@ -1,8 +1,11 @@
+import { pipe, pipeWith } from 'pipe-ts';
 import * as urlHelpers from 'url';
 import { UrlObject, UrlWithParsedQuery, UrlWithStringQuery } from 'url';
 import { getOrElseMaybe, mapMaybe } from './helpers/maybe';
 import { flipCurried, isNonEmptyString } from './helpers/other';
-import { pipe } from './helpers/pipe';
+
+type Update<T> = T | ((prev: T) => T);
```

Comment:
I think we should only allow update functions, as they are more flexible and if you don't need to access the previous value, it's not too much trouble to wrap your value in a function anyway. However, this would be a large breaking change so I'm putting it off for now.

Reply:
> However, this would be a large breaking change so I'm putting it off for now.

It's worth noting we have the comfort of introducing breaking changes, and if they are to be introduced, then the best time is now.

The current version of this library is `0.0.6`, so nowhere close to stable. Also, GitHub stats say that as of this moment, the library has only a few users they know of. If a breaking change is justified, it's better to do it before it's too late.

To ease out any potential pain, the value could be wrapped in a thunk _by_ the library if necessary, and a deprecation notice could be displayed.

```ts
if (!(newPath instanceof Function)) {
    console.warn('Deprecation warning: expected "newPath" to be a function. See https://github.com/unsplash/url-transformers#usage.')
    path = () => newPath;
}
```

Reply:
I'm not really worried about breaking external users—as you say, it's a v0 library, so breaking changes are expected.

It's more that right now I don't want to spend time updating all of our own usages. I see that change as orthogonal to the one I'm trying to make here (allow access to the previous value), and I don't want the scope to creep beyond that.

I would like to make this change alongside other breaking changes, e.g. #9.

Reply:
P.S. this change is now done alongside other breaking changes in #9